swarmlib/util/functions.py

```diff
@@ -28,7 +31,19 @@ def ackley(x, a=20, b=0.2, c=2*np.pi):
     return -a*np.exp(-b*np.sqrt(s1 / n)) - np.exp(s2 / n) + a + np.exp(1)
 
 
+# Wrapper for landscapes.single_objective functions for inputs > 1d
+def wrap_landscapes_func(landscapes_func):
```

Comment:
Why do we need the wrapper? Can't we just call landscapes 2D functions directly?

Reply:
On initialization the `BaseVisualiser` class calls the objective function with a 3D array [here](https://github.com/alxfmpl/swarmlib/blob/7eadfb7c6632a1b81c90040e3ac31b6cd9938475/swarmlib/util/base_visualizer.py#L41). `landscapes` only accepts 1d function calls.

Reply:
Ok I see. Because landscape does not use numpy. Then this is fine for me

Reply:
@alxfmpl maybe we can try to use the `@np.vectorize` decorator. This could lead to somewhat cleaner code. I found this [SO post](https://stackoverflow.com/a/20228722/6925187).

Reply:
Could try that, it would lead to something like [this](https://stackoverflow.com/questions/3467526/attaching-a-decorator-to-all-functions-within-a-class). Definitely looks quite clean, maybe not as readable. I'll give it a shot.

Reply:
Okay after having a look at it I don't think this is applicable in this case. `np.vectorize` makes nested inputs possible, which is similar to `np.apply_along_axis(func1d=landscapes_func, axis=2, arr=x)` when`x` is a 3D array. I couldn't come up with a way around this...

Reply:
OK, thx for trying it anyway. Then we are good to merge :)